sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -89,9 +89,9 @@ def on_request(self, request):
         """
         self._enforce_https(request)
 
-        if self._need_new_token:
+        if self._token is None or self._need_new_token:
             self._token = self._credential.get_token(*self._scopes)
```

Comment:
def _need_new_token(self):
        # type: () -> bool
        return not self._token or self._token.expires_on - time.time() < 300

Seems like self._token is None is not needed?

Reply:
It's unnecessary (see my PR description), but that's the only way to make mypy happy